backend/src/main/java/com/handong/rebon/shop/domain/tag/application/TagService.java

```diff
@@ -1,7 +1,55 @@
 package com.handong.rebon.shop.domain.tag.application;
 
+import com.handong.rebon.exception.tag.TagExistException;
+import com.handong.rebon.shop.domain.tag.application.dto.TagRequestDto;
+import com.handong.rebon.shop.domain.tag.domain.Tag;
+import com.handong.rebon.shop.domain.tag.domain.repository.TagRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
 
 @Service
+@Transactional
+@RequiredArgsConstructor
 public class TagService {
+
+    @Autowired
+    TagRepository tagRepository;
+
+    public Long createTag(String name) {
+        validateDuplicateTag(name);
+
+        Tag newTag = Tag.builder()
+                .name(name)
+                .build();
+
+        tagRepository.save(newTag);
+        return newTag.getId();
+    }
+
+    public Long createTag(TagRequestDto tagRequestDto) {
+        String tagName = tagRequestDto.getName();
+        validateDuplicateTag(tagName);
+
+        Tag newTag = Tag.builder()
+                .name(tagName)
+                .build();
+
+        tagRepository.save(newTag);
+        return newTag.getId();
```

Comment:
이렇게 해줘도 JPA에 의해 id값이 들어갈 것 같긴하지만 가독성을 위해 새로 반환되는 값을 사용하는 건 어떨까요??
```suggestion
        Tag savedTag = tagRepository.save(newTag);
        return savedTag.getId();
```

Reply:
이렇게 보니, 반환 값을 받는게 훨씬 좋아보이네요! 감사합니다!